cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -246,7 +248,226 @@ EXEC strQry // SQL Injection vulnerability!
+ 
+//check to make sure an ip address was provided    
+if (string.IsNullOrEmpty(address))  
+{   
+	try
+	{
+		// Create an instance of IPAddress for the specified address string (in 
+		// dotted-quad, or colon-hexadecimal notation).
+		IPAddress address = IPAddress.Parse(ipAddress);
+
+		// Display the address in standard notation.
+		return address.ToString();
+	}
+	catch(ArgumentNullException e)
+	{
+		//ipaddress is not of type IPaddress
+		...
+	}
+    ...
+}
+ ```
+ 
+### LDAP injection
+
+Almost any characters can be used in Distinguished Names. However, some must be escaped with the backslash "\" escape character. Active Directory requires that the following characters be escaped:
+
+| Character | Escape character |
+|-----------|:-----:|
+|Comma	|,|
+|Backslash character|\|
+|Pound sign (hash sign)|#|
+|Plus sign|+|
+|Less than symbol|<|
+|Greater than symbol|>|
+|Semicolon|;|
+|Double quote (quotation mark)|"|
+|Equal sign|=|
+|Leading or trailing spaces| | 
+
+NB: The space character must be escaped only if it is the leading or trailing character in a component name, such as a Common Name. Embedded spaces should not be escaped.
+
+## A2 Broken Authentication
+
+DO: Use [ASP.net Core Identity](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity?view=aspnetcore-2.2&).
+ASP.net Core Identity framework is well configured by default, where it uses secure password hashes and an individual salt. Identity uses the PBKDF2 hashing function for passwords, and they generate a random salt per user.
+
+DO: Set secure password policy
+
+e.g ASP.net Core Identity 
+
+``` csharp
+//startup.cs
+services.Configure<IdentityOptions>(options =>
+{
+	// Password settings
+	options.Password.RequireDigit = true;
+	options.Password.RequiredLength = 8;
+	options.Password.RequireNonAlphanumeric = true;
+	options.Password.RequireUppercase = true;
+	options.Password.RequireLowercase = true;
+	options.Password.RequiredUniqueChars = 6;
+ 
+ 
+	options.Lockout.DefaultLockoutTimeSpan = TimeSpan.FromMinutes(30);
+	options.Lockout.MaxFailedAccessAttempts = 3;
+ 
+	options.SignIn.RequireConfirmedEmail = true;
+	
+	options.User.RequireUniqueEmail = true;
+});
+```
+
+DO: Set a cookie policy
+
+e.g 
+
+``` csharp
+//startup.cs
+services.ConfigureApplicationCookie(options =>
+{
+	options.Cookie.HttpOnly = true;
+	options.Cookie.Expiration = TimeSpan.FromHours(1)
+	options.SlidingExpiration = true;
+});
+```
+
+## A3 Sensitive Data Exposure
+
+DO NOT: [Store encrypted passwords](Password_Storage_Cheat_Sheet.md#do-not-limit-the-character-set-and-set-long-max-lengths-for-credentials).
+
+DO: Use a strong hash to store password credentials. Use Argon2, PBKDF2, BCrypt or SCrypt with at least 8000 iterations and a strong key.
+
+DO: Enforce passwords with a minimum complexity that will survive a dictionary attack i.e. longer passwords that use the full character set (numbers, symbols and letters) to increase the entropy.
```

Comment:
Same remark like above but to relative linking to [this section](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Password_Storage_Cheat_Sheet.md#do-not-limit-the-character-set-and-set-long-max-lengths-for-credentials).

Reply:
Still missing when I take the content of your [fork](https://github.com/knadt/CheatSheetSeries/blob/master/cheatsheets/DotNet_Security_Cheat_Sheet.md)